src/dataMigration.js

```diff
@@ -52,4 +54,18 @@ const dataMigrations = [
       if (syncURL && syncURL.length > 0) settings.set(SETTINGS_KEYS.SYNC_IS_INITIALISED, 'true');
     },
   },
+  {
+    version: '2.0.0',
+    migrate: database => {
+      // Changed SyncQueue to expect no more than one SyncOut record for every record in database.
+      // Assume that last SyncOut record is correct.
+      const allRecords = database.objects('SyncOut').sorted('changeTime');
+      database.write(() => {
+        allRecords.forEach(record => {
+          const hasDuplicates = allRecords.filtered('recordId == $0', record.id).length > 0;
+          if (hasDuplicates) database.delete('SyncOut', record);
+        });
+      });
+    },
+  },
```

Comment:
This is nice, but it isn't enough. If there's a transaction that's been updated once, then had lines added and then finalised, the sync queue will still be out of order after this migration code is run. Soz.